File: Suraj/Sorting/insertion_sort.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> insertionSort(vector<int>& nums) {
        int n = nums.size();
        for(int i=0;i<n;i++){
            int j = i;
            while(j>0 && nums[j-1]>nums[j]){
                swap(nums[j-1],nums[j]);
                j--;

            }
        }
        return nums;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {5,2,3,1};
    vector<int> ans = sol.insertionSort(nums);
    for(auto x: ans){
        cout<<x<<" ";
    } 
    return 0;
}

File: Ansh/sorting/insertionSort.py
class Solution:
    def insertionSort(self, nums):
        n=len(nums)
        for i in range(n):
            for j in range(i,0,-1):
                if(nums[j]<nums[j-1]):
                    nums[j],nums[j-1]=nums[j-1],nums[j]
        return nums
if __name__=="__main__":
    nums=[64, 25, 12, 22, 11]
    s=Solution()
    print(s.insertionSort(nums))
    # by ANSH PATERIYA
    